src/rgw/rgw_aws_lambda.cc

```diff
@@ -0,0 +1,209 @@
+      std::cout << "Invoking.." << std::endl;
+      const Aws::Lambda::LambdaClient *client = msg->client;
+      Aws::Lambda::Model::InvokeRequest invokeRequest;
+      invokeRequest.SetInvocationType(Aws::Lambda::Model::InvocationType::RequestResponse);
+      invokeRequest.SetLogType(Aws::Lambda::Model::LogType::Tail);
+      invokeRequest.SetFunctionName(Aws::String(msg->function_name));
+      std::shared_ptr<Aws::IOStream> body = Aws::MakeShared<Aws::StringStream>("Simple Allocation Tag");
+      *body << msg->payload;
+      invokeRequest.SetBody(body);
+      invokeRequest.SetContentType("application/json");
+      auto response = client->Invoke(invokeRequest);
+      if (!response.IsSuccess()) {
+        std::cout << response.GetError().GetMessage() << std::endl;
+      } else {
+        Aws::Lambda::Model::InvokeResult &result = response.GetResult();
+        Aws::IOStream &pl = result.GetPayload();
+        Aws::String returnedData;
+        std::getline(pl, returnedData);
+        std::cout << "Success " << returnedData << "\n" << std::endl;
+      }
+    }
+
+//    Works on messages in the queque
+    void run() {
+      while (!stopped) {
+        const auto count = messageQueue.consume_all(
+                std::bind(&Manager::publish_internal, this, std::placeholders::_1));
+        if (count == 0) {
+          std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+        }
+      }
+    }
+
+  public:
+    Manager(size_t _max_clients, size_t max_queue_size, CephContext *_cct) : messageQueue(max_queue_size),
+                                                                             runner(&Manager::run, this),
+                                                                             stopped(false), cct(_cct),
+                                                                             clientList(_max_clients) {}
+
+    Aws::Lambda::LambdaClient *connect(const std::string &accessKey,
+                                       const std::string &accessSecret,
+                                       const std::string &arn,
+                                       const std::string &caPath,
+                                       bool useSSL) {
+      if (stopped) {
+        return nullptr;
+      }
+      const client_id_t id(accessKey, accessSecret, arn, useSSL);
+      std::lock_guard lock(clientsLock);
+      const auto it = clientList.find(id);
+      if (it != clientList.end()) {
+        return it->second;
+      }
+
+      // connection not found, creating a new one
+      Aws::Client::ClientConfiguration configuration;
+      if (useSSL) {
+        configuration.scheme = Aws::Http::Scheme::HTTPS;
+        configuration.verifySSL = true;
+      } else {
+        configuration.scheme = Aws::Http::Scheme::HTTP;
+        configuration.verifySSL = false;
+      }
+      Aws::Utils::ARN arn1((Aws::String(arn)));
+      configuration.region = Aws::String(arn1.GetRegion());
+      configuration.maxConnections = 25;
```

Comment:
please make these constants, const data members of the Manager class (you can initialize them in the Manager ctor).
this will make it easier to take them from CephContext conf in the future